ludwig/utils/metric_utils.py

```diff
@@ -89,13 +91,24 @@ def reduce_trainer_metrics_dict(
     return {k: dict(v) for k, v in flattened_dict.items()}
 
 
-def get_metric_names(output_features: Dict[str, Dict]) -> Dict[str, List[str]]:
+def get_metric_names(output_features: Dict[str, "OutputFeature"]) -> Dict[str, List[str]]:  # noqa
     """Returns a dict of output_feature_name -> list of metric names."""
     metrics_names = {}
     for output_feature_name, output_feature in output_features.items():
-        for metric in output_feature.metric_functions:
-            metrics = metrics_names.get(output_feature_name, [])
-            metrics.append(metric)
-            metrics_names[output_feature_name] = metrics
+        metrics_names[output_feature_name] = sorted(
+            list(get_metric_feature_type_registry()[output_feature.type()].keys())
+        )
+    # Add combined loss.
+    metrics_names[COMBINED] = [LOSS]
+    return metrics_names
+
+
+def get_feature_to_metric_names_map(output_features: List[FeatureConfigDict]) -> Dict[str, List[str]]:
+    """Returns a dict of output_feature_name -> list of metric names."""
+    metrics_names = {}
+    for output_feature in output_features:
+        output_feature_name = output_feature[NAME]
+        output_feature_type = output_feature[TYPE]
+        metrics_names[output_feature_name] = get_metric_feature_type_registry()[output_feature_type]
```

Comment:
Suggestion: A lot of the other registries have accessor methods (i.e. `get_encoder_classes`). It may be worthwhile to add a similar accessor method here for the `metric_feature_type_registry` to remain consistent and make it a little clearer what you're actually getting on this line of code.